### PR TITLE
feat: add BakeShadows component

### LINKED ISSUE #321: BakeShadows
### Description

I would like to have a simple way to baked my shadows using threeJs and the shadow.autoUpdate function on the render as false

### Suggested solution

To create a simple renderless doing this

### Alternative

_No response_

### Additional context

_No response_

### Validations

- [X] I agree to follow this project's [Code of Conduct](https://github.com/Tresjs/cientos/blob/main/CODE_OF_CONDUCT.md)
- [X] Read the [Contributing Guidelines](https://github.com/Tresjs/cientos/blob/main/CONTRIBUTING.md).
- [X] Read the [docs](https://cientos.tresjs.org/guide).
- [X] Check that there isn't [already an issue](https://github.com/tresjs/cientos/issues) that reports the same bug to avoid creating a duplicate.

### PR DESCRIPTION
**Pull request by JaimeTorrealba**
closes #321  
How can I remove these warnings

image

### CODE REVIEW

File: src/core/misc/BakeShadows.ts
Comment (alvarosabu) [inline src/core/misc/BakeShadows.ts:1]: @JaimeTorrealba probably because you are defining any props

Comment (JaimeTorrealba) [reply src/core/misc/BakeShadows.ts:1]: But <Stats /> has props, and the messages still appears

**Comment by andretchen0** [comment]
About the color error, the `color` prop on `TresMeshStandardMaterial` expects a `THREE.Color`, so any of [the instantiation examples here](https://threejs.org/docs/#api/en/math/Color) will work, like this:

```
<TresMeshStandardMaterial :color="new Color(0x00ff00)" />
```

File: src/core/misc/BakeShadows.ts [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+
+  setup() {
+    const { renderer } = useTresContext()
+
+    watchEffect(() => {
+      renderer.value.shadowMap.autoUpdate = false
+      renderer.value.shadowMap.needsUpdate = true
+    })
+
+    return null
```
Comment (andretchen0) [inline src/core/misc/BakeShadows.ts:15]: Remove `return null`. I believe this fixes the typing issue. Let me know if that doesn't fix it for you.

[Fwiw, here's an example with no returned value in the Vue docs.](https://vuejs.org/guide/typescript/composition-api.html#without-script-setup)

File: playground/src/pages/misc/BakeShadows.vue [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+  >
+    <TresPerspectiveCamera :position="[0, 2, 5]" />
+    <CameraControls />
+    <BakeShadows />
+    <TresMesh
+      ref="cubeRef"
+      cast-shadow
+    >
+      <TresBoxGeometry />
+      <TresMeshStandardMaterial :color="0x00ff00" />
```
Comment (andretchen0) [inline playground/src/pages/misc/BakeShadows.vue:37]: `TresMeshStandardMaterial`'s `color` prop expects a `THREE.Color`, so you'll have to instantiate one with `new Color(...)`.

File: playground/src/pages/misc/BakeShadows.vue [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+    </TresMesh>
+    <TresMesh
+      receive-shadow
+      :position="[0, -2, 0]"
+      :rotation-x="-Math.PI / 2"
+    >
+      <TresPlaneGeometry :args="[5, 5]" />
+      <TresMeshStandardMaterial :color="0xf7f7f7" />
```
Comment (andretchen0) [inline playground/src/pages/misc/BakeShadows.vue:45]: Same as above. This should be a `THREE.Color`, not a number.

**Review by andretchen0 — CHANGES_REQUESTED** [review]

File: src/core/misc/BakeShadows.ts (hunk 1 above)
Comment (JaimeTorrealba) [reply src/core/misc/BakeShadows.ts]: Thank you so much, it was that 🙏

File: playground/src/pages/misc/BakeShadows.vue (hunk 3 above)
Comment (JaimeTorrealba) [reply playground/src/pages/misc/BakeShadows.vue]: Same here

File: playground/src/pages/misc/BakeShadows.vue (hunk 2 above)
Comment (JaimeTorrealba) [reply playground/src/pages/misc/BakeShadows.vue]: Done :)

**Review by andretchen0 — APPROVED** [review]
Looks good to me!

**Pull request merged**